validator/db/kv/attestation_history_v2.go

```diff
@@ -42,81 +43,108 @@ func (hd EncHistoryData) assertSize() error {
 	return nil
 }
 
-func newAttestationHistoryArray(target uint64) EncHistoryData {
-	enc := make(EncHistoryData, latestEpochWrittenSize+(target%params.BeaconConfig().WeakSubjectivityPeriod)*historySize+historySize)
+func (h *HistoryData) IsEmpty() bool {
+	if h == (*HistoryData)(nil) {
+		return true
+	}
+	if h.Source == params.BeaconConfig().FarFutureEpoch {
+		return true
+	}
+	return false
+}
+
+func emptyHistoryData() *HistoryData {
+	h := &HistoryData{Source: params.BeaconConfig().FarFutureEpoch, SigningRoot: bytesutil.PadTo([]byte{}, 32)}
+	return h
+}
+
+// NewAttestationHistoryArray creates a new encapsulated attestation history byte array
+// sized by the latest epoch written.
+func NewAttestationHistoryArray(target uint64) *EncHistoryData {
+	en := make(EncHistoryData, latestEpochWrittenSize+(target%params.BeaconConfig().WeakSubjectivityPeriod)*historySize+historySize)
```

Comment:
please refactor this line, 4 different operations are happening here at once.

Reply:
refactored